src/initialise_db.py

```diff
@@ -0,0 +1,106 @@
+'''
+Copyright (C) 2020 Link Shortener Authors
+Licensed under the MIT (Expat) License. See the LICENSE file found in the
+top-level directory of this distribution.
+'''
+from sanic import Blueprint
+
+from aiomysql import create_pool
+from aiomysql.sa import create_engine
+
+from sqlalchemy import MetaData, Table, Column, String, Integer, Boolean
+from sqlalchemy.schema import CreateTable
+
+
+initdb_blueprint = Blueprint('intitialise_db')
+
+metadata = MetaData()
+initdb_blueprint.table = Table(
+    'links',
+    metadata,
+    Column('owner', String(50)),
+    Column('endpoint', String(20), unique=True),
```

Comment:
Regarding this `unique=True`... In the future, it would be nice if we support reusing of the endpoints. E.g.:

- In 2020, we would like to create a redirect `https://fueled.by/hackers -> https://www.applifting.com/hackathon2020`.
- In 2021, we would like to deactivate this redirect.
- In 2022, we would like to create another redirect `https://fueled.by/hackers -> https://www.applifting.com/hackathon2022`.

With this constraint that won't be possible.

One of the possible options is to define the unique constraint on both columns together like: `UniqueConstraint('endpoint', 'url')`. On the other hand, this constraint allows us to have multiple **active** occurences of the same `endpoint` in a single moment. I wonder how can we avoid that. Of course, we can put this logic into the code instead of the database constraints. Or maybe we can move inactive links into another table... :thinking:

Reply:
Constraint removed, then put back to active links table.